src/RunState.cc

```diff
@@ -414,20 +414,38 @@ void get_final_stats()
 		{
 		iosource::PktSrc::Stats s;
 		ps->Statistics(&s);
-		double dropped_pct = s.dropped > 0.0
-		                         ? ((double)s.dropped / ((double)s.received + (double)s.dropped)) *
-		                               100.0
-		                         : 0.0;
+
+		double dropped_pct = 0.0;
+		if ( s.dropped > 0 )
+			dropped_pct = (static_cast<double>(s.dropped) /
+			               (static_cast<double>(s.received) + static_cast<double>(s.dropped))) *
+			              100.0;
 
 		uint64_t not_processed = packet_mgr->GetUnprocessedCount();
 		double unprocessed_pct = not_processed > 0
 		                             ? ((double)not_processed / (double)s.received) * 100.0
 		                             : 0.0;
 
-		reporter->Info("%" PRIu64 " packets received on interface %s, %" PRIu64
-		               " (%.2f%%) dropped, %" PRIu64 " (%.2f%%) not processed",
-		               s.received, ps->Path().c_str(), s.dropped, dropped_pct, not_processed,
-		               unprocessed_pct);
+		if ( s.filtered )
+			{
+			double filtered_pct = 0.0;
+			if ( s.filtered && s.filtered.value() > 0 )
+				filtered_pct = (static_cast<double>(s.filtered.value()) /
+				                (static_cast<double>(s.received) +
+				                 static_cast<double>(s.filtered.value()))) *
+				               100.0;
+
+			reporter->Info("%" PRIu64 " packets received on interface %s, %" PRIu64
```

Comment:
I probably would've tried to keep the single `reporter->Info()` invocation with a possibly empty filtered_msg suffix formatted in. Should be less duplicated and less lines.
```
std::string filtered_msg = "";
if ( s.filtered )
   filtered_msg = zeek::util::fmt(" ...filtered")

reporter->Info(... %s", ..., filtered_msg.c_str());
```